internal/k8shandler/reconciler.go

```diff
@@ -34,17 +34,17 @@ func SecretReconcile(requestCluster *elasticsearchv1.Elasticsearch, requestClien
 	newSecretHash := getSecretDataHash(requestCluster.Name, requestCluster.Namespace, requestClient)
 	// compare the new secret with current one in the nodes
 	for _, node := range nodes[nodeMapKey(requestCluster.Name, requestCluster.Namespace)] {
-		if newSecretHash != node.GetSecretHash() {
+		if newSecretHash != node.getSecretHash() {
 			secretChanged = true
+
+			_, nodeStatus := getNodeStatus(node.name(), &requestCluster.Status)
+			// requestCluster.Status.Nodes[idx].UpgradeStatus.ScheduledForCertRedeploy = corev1.ConditionTrue
+			nodeStatus.UpgradeStatus.ScheduledForCertRedeploy = corev1.ConditionTrue
 		}
 	}
 
 	if secretChanged {
 		// Cluster's secret has been updated, set the nodes to be redeployed
-		for _, node := range requestCluster.Status.Nodes {
-			node.UpgradeStatus.ScheduledForCertRedeploy = corev1.ConditionTrue
-		}
-
 		err := requestClient.Status().Update(context.TODO(), requestCluster)
```

Comment:
we should probably wrap this in a `retry.RetryOnConflict` function similar to what is done in `status.go:L72-97`